### ISSUE TITLE
Implement rolling updates

### ISSUE THREAD

**Issue by mpldr** [tachiyomiorg/tachiyomi#7003]
### Describe your suggested feature

Every time an update runs (even when it's in Background) I am reminded that "large updates harm sources…". As a way to reduce strain on sources and only sending this toast on manual "complete updates", I'd suggest "rolling" updates (no idea how to phrase that differently). Meaning: only query for updates for a few manga at once every x units of time.

As an example:
Assuming we have 12 Manga at one source:
```
15:00: 1, 2, 3 updated
15:05: 4, 5, 6 updated
15:10: (internet connection cuts out/update interrupted) 8 updated
15:15: 7, 9, 10 updated
15:20: 11, 12 updated
```

This would flatten the curve for the source and would theoretically even allow for more frequent update (twice a day is still plenty though). If multiple sources are queried, these can be done in parallel (with a limit on how many at the same time to limit strain on the phone CPU/Network), like:
```
15:00: S1-1, S1-2, S1-3, S2-1, S2-2 updated
15:05: S1-4, S1-5, S1-6, S2-3, S2-4 updated
15:10: S2-5 updated
```

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have updated the app to version **[0.13.4](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by arkon** [comment]
Similar to #6968. Feel free to add a comment there to fill in the gaps if any.
